libnm-core/nm-keyfile.c

```diff
@@ -2896,6 +2898,9 @@ nm_keyfile_read (GKeyFile *keyfile,
 	info.user_data = user_data;
 
 	groups = g_key_file_get_groups (keyfile, &length);
+	if (!groups)
+		length = 0;
+
```

Comment:
g_key_file_get_*_list() always set the length, while g_key_file_get_keys() does not; this is a bit inconsistent in glib. Anyway I think it's correct not to assume that the length is set because the function can return an error.

OTOH, g_key_file_get_groups () always sets _length_ and can't fail so I think it's safe to rely on _length_ value.

Reply:
hm, that's true. In several cases, it's not a severe bug.
- not severe for `g_key_file_get_groups()` as it never fails
- not severe for `g_key_file_get_*_list()` as they actually set the length (undocumented)
- not severe for our call of `g_key_file_get_keys()`, because in the circumstances we call the function it's not expected to fail.

Anyway, I'd keep all the changes there, simply to be explicit about it. It doesn't hurt (and it was only partly clear to me that it is in fact not needed). I'd update the commit message though, stating that it the issue is not severe.

Reply:
> Anyway, I'd keep all the changes there, simply to be explicit about it. It doesn't hurt (and it was only partly clear to me that it is in fact not needed). I'd update the commit message though, stating that it the issue is not severe.

Okay.